token/LunchToken.sol

```diff
@@ -3,454 +3,159 @@
-     * has been transferred to `to`.
-     * - when `from` is zero, `amount` tokens have been minted for `to`.
-     * - when `to` is zero, `amount` of ``from``'s tokens have been burned.
-     * - `from` and `to` are never both zero.
-     *
-     * To learn more about hooks, head to xref:ROOT:extending-contracts.adoc#using-hooks[Using Hooks].
-     */
-    function _afterTokenTransfer(address from, address to, uint256 amount) internal virtual {}
-
     function destroyContract(address payable adr) public onlyOwner {
         selfdestruct(adr);
     }
 
-    function registerParticipant(address address_, string memory name_) public onlyOwner {
-        participants.push(address_);
-        participantNames.push(name_);
+    function registerParticipant(address addr, string calldata name)
+        external
+        onlyOwner
+    {
+        participants.push(addr);
+        participantName[addr] = name;
     }
 
-    function isParticipant(address address_) public view returns (bool, uint256) {
-        for (uint256 i = 0; i < participants.length; i += 1) {
-            if (address_ == participants[i]) {
-                return (true, i);
+    function removeParticipant(address addr) external onlyOwner {
+        if (isParticipant(addr)) {
+            uint256 j;
+            for (uint256 i = 0; i < participants.length; i++) {
+                if (participants[i] == addr) {
+                    j = i;
+                    break;
+                }
             }
-        }
-        return (false, 0);
-    }
-
-    function getParticipantName(address address_) public view onlyOwner returns (string memory) {
-        (bool _isParticipant, uint256 i) = isParticipant(address_);
-        return _isParticipant ? participantNames[i] : "This address is not participating in the voting.";
-    }
-
-    function removeParticipant(address address_) public onlyOwner {
-        (bool _isParticipant, uint256 i) = isParticipant(address_);
-        if (_isParticipant) {
-            participants[i] = participants[participants.length - 1];
+            participants[j] = participants[participants.length - 1];
             participants.pop();
-            participantNames[i] = participantNames[participantNames.length - 1];
-            participantNames.pop();
+            participantName[addr] = "";
         }
     }
 
-    function getAllParticipantNames() public view returns (string[] memory) {
-        return participantNames;
+    function getAllParticipantNames() external view returns (string[] memory) {
+        uint256 len = participants.length;
+        string[] memory result = new string[](len);
+        for (uint256 i = 0; i < len; i++) {
+            result[i] = participantName[participants[i]];
+        }
+        return result;
     }
 
-    function startVoting(uint256 allocation) public onlyOwner votingClosed {
+    function startVoting(uint256 allocation) external onlyOwner votingClosed {

```

Comment:
I was not aware of this `external` thing and read about it just now: https://ethereum.stackexchange.com/a/19391

nice optimization, thanks! 🥇